kostyor/tests/unit/upgrades/test_engine.py

```diff
@@ -315,107 +280,84 @@ def test_all_in_one_assignment(self):
+            mock.call({'name': 'neutron-server',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'neutron-linuxbridge-agent',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'neutron-l3-agent',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'neutron-dhcp-agent',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'neutron-metering-agent',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'neutron-metadata-agent',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'neutron-ns-metadata-proxy',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'cinder-api',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'cinder-scheduler',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'cinder-volume',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'heat-api',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'heat-engine',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'heat-api-cfn',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'heat-api-cloudwatch',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
         ]
         self.assertEqual(
-            expected_service_calls,
-            self.engine.driver.pre_service_upgrade_hook.call_args_list)
-        self.assertEqual(
-            expected_service_calls,
-            self.engine.driver.start_upgrade.call_args_list)
-        self.assertEqual(
-            expected_service_calls,
-            self.engine.driver.post_service_upgrade_hook.call_args_list)
-
-        expected_host_calls = [
-            mock.call(self.upgrade,
-                      {'hostname': 'host-1',
-                       'cluster_id': '2ba8fad8-3a0f-47db-a45b-62df1d811687',
-                       'id': '5708c51f-5421-4ecd-9a9e-000000000001'}),
-        ]
-        self.assertEqual(
-            expected_host_calls,
-            self.engine.driver.pre_host_upgrade_hook.call_args_list)
-        self.assertEqual(
-            expected_host_calls,
-            self.engine.driver.post_host_upgrade_hook.call_args_list)
+            expected_start_calls,
```

Comment:
It's a good question. :) `assert_has_calls` does not check strictly, i.e. it checks that the calls are happened in a given order but if there were any other calls - that's fine. In our case I want to ensure no other calls have been made except the ones I'm expecting.